am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/RevokeAccessIntegrationTest.java

```diff
@@ -35,6 +44,83 @@ void whenRevokingResourceAccessResourceAccessRemovedFromDatabase() {
         assertThat(countResourcesById(resourceId)).isEqualTo(0);
     }
 
+    @Test
+    void whenRevokingResourceAccessOnRootResourceAccessRemovedFromDatabase() {
+        ams.grantExplicitResourceAccess(createGrantForWholeDocument(resourceId, READ_PERMISSION));
+        ams.revokeResourceAccess(ExplicitAccessMetadata.builder()
+            .resourceId(resourceId)
+            .accessorId(ACCESSOR_ID)
+            .accessType(ACCESS_TYPE)
+            .serviceName(SERVICE_NAME)
+            .resourceType(RESOURCE_TYPE)
+            .resourceName(RESOURCE_NAME)
+            .attribute("/")
+            .securityClassification(SECURITY_CLASSIFICATION)
+            .build());
+
+        assertThat(countResourcesById(resourceId)).isEqualTo(0);
```

Comment:
Please add new test to grant access to Resource 1 and Resource 2 (two unique resources) and make sure that when you revoke access to let's say Resource 1 then rules for Resource 2 remain.

Reply:
👍